src/core/src/OmniTileset.cpp

```diff
@@ -485,47 +493,53 @@ void OmniTileset::reload() {
     }
 
     const auto cesiumTileset = UsdUtil::getCesiumTileset(_pContext->getUsdStage(), _path);
-    for (const auto& child : cesiumTileset.GetPrim().GetAllChildren()) {
-        const auto rasterOverlayPath = child.GetPath();
-        if (UsdUtil::isCesiumRasterOverlay(_pContext->getUsdStage(), rasterOverlayPath)) {
-            _rasterOverlayPaths.push_back(rasterOverlayPath);
-        }
+    const auto rasterOverlayPaths = getRasterOverlayPaths();
 
+    for (const auto& rasterOverlayPath : rasterOverlayPaths) {
         const auto pRasterOverlay = _pContext->getAssetRegistry().getRasterOverlay(rasterOverlayPath);
         if (pRasterOverlay) {
-            const auto pCesiumRasterOverlay = pRasterOverlay->getRasterOverlay();
-            if (pCesiumRasterOverlay) {
-                _pTileset->getOverlays().add(pCesiumRasterOverlay);
+            const auto pNativeRasterOverlay = pRasterOverlay->getRasterOverlay();
+            if (pNativeRasterOverlay) {
+                _pTileset->getOverlays().add(pNativeRasterOverlay);
             }
         }
     }
 }
 
-std::optional<uint64_t>
-OmniTileset::getRasterOverlayLayerIndex(const CesiumRasterOverlays::RasterOverlay& overlay) const {
-    uint64_t rasterOverlayLayerIndex = 0;
-    for (const auto& pOverlay : _pTileset->getOverlays()) {
-        if (pOverlay.get() == &overlay) {
-            return rasterOverlayLayerIndex;
-        }
+pxr::SdfPath OmniTileset::getRasterOverlayPath(const CesiumRasterOverlays::RasterOverlay& rasterOverlay) const {
+    const auto rasterOverlayPaths = getRasterOverlayPaths();
 
-        ++rasterOverlayLayerIndex;
+    for (const auto& rasterOverlayPath : rasterOverlayPaths) {
+        const auto pRasterOverlay = _pContext->getAssetRegistry().getRasterOverlay(rasterOverlayPath);
+        if (pRasterOverlay) {
+            const auto pNativeRasterOverlay = pRasterOverlay->getRasterOverlay();
+            if (pNativeRasterOverlay == &rasterOverlay) {
+                return rasterOverlayPath;
+            }
+        }
     }
 
-    return std::nullopt;
+    return {};
 }
 
-uint64_t OmniTileset::getRasterOverlayLayerCount() const {
-    return _rasterOverlayPaths.size();
-}
+void OmniTileset::updateRasterOverlayAlpha(const pxr::SdfPath& rasterOverlayPath) {
+    const auto rasterOverlayPaths = getRasterOverlayPaths();
+    const auto rasterOverlayIndex = CppUtil::indexOf(rasterOverlayPaths, rasterOverlayPath);
+
+    if (rasterOverlayIndex == rasterOverlayPaths.size()) {
```

Comment:
Any case where `>=` would be true but not `==`?

Reply:
It shouldn't ever happen. Kind of like how `std::find` returns the end iterator if not found, `indexOf` returns the size of the vector.